### ISSUE TITLE
Warning when using keyword from nested library.

### ISSUE THREAD

**Issue by Mario-P** [nokia/RED#50]
There is warning displayed when keywords from nested library are used:
![warning](https://cloud.githubusercontent.com/assets/22638505/19393762/6c0f0a56-9236-11e6-9db4-d6a575251687.PNG)
Project structure:
![structure](https://cloud.githubusercontent.com/assets/22638505/19393775/74febf80-9236-11e6-86a3-f9a12a7ebd31.PNG)

Is there anything that can be done not to have such warnings?

**Comment by adam-wypych** [comment]
We have such enhancement in our backlog, but currently such option is not available. We will check possibility to implement is ASAP.

**Comment by KrzysztofJozefowicz** [comment]
I moved your idea somewhere at the top as it was also requested from internal clients thus it looks that it would be used regularly. First we need to fix reported bugs.

**Comment by KrzysztofJozefowicz** [comment]
This will be provided in 0.7.3 release.

**Comment by Mario-P** [comment]
It is still the same in 0.7.3.

**Comment by KrzysztofJozefowicz** [comment]
Yes, this error/warning is still 0.7.3 as the problem is valid one. What 0.7.3 allows is to suppress such issues.
Go to Windows -> Preferences -> Robot Framework -> Errors/Warnings and under Code styles  you will find "Keyword from nested library" with default value "warning". You can change it to info or ignore if you need to.
After the change you can revalidate file or whole project by ctrl+F5.

**Comment by humbienri** [comment]
Hello @KrzysztofJozefowicz, first off thank you guys for all of this work that you've gifted us!  It's invaluable!  I am wondering what you mean by "problem is valid one".  Why is it valid? Or how is it that it's a problem?  Is it an anti-pattern that one should avoid?  I am new to RED and am enjoying using it although I need to figure out how to make Eclipse (portable version) more responsive and snappy.  On my laptop it does act a little "choppy."

**Comment by michalanglart** [comment]
Hi @humbienri 

I think the opinions vary depending on user so I'm not going to tell you if that's anti-pattern or a pattern. As to myself I would prefer to keep dependencies explicitly defined rather than implicitely by transitive relations. Once you have such transitive dependencies you need some good code navigation support as it would be harder to navigate between files in e.g. some simple notepad-based environments.

> On my laptop it does act a little "choppy."

What do you mean by that?

**Comment by adam-wypych** [comment]
I totally agree with @michalanglart that is not possible to make correct judgment about it is pattern or anti-pattern without view on concrete usage. From my experience, I can only say, that usage of library from resource file only creates problem (at least with readability) if nesting level (starting form test suite file) will be greater than `1` or amount of libraries inside such file is equal to `1` (because it is totally useless and you can inline library import directly in every test suite).

So in other words if you are using your resource file pretty much as `__init__.py` is generally fine.

For more see below links:
[1] https://stackoverflow.com/questions/48524292/robot-framework-multiple-resource-files-accessing-the-same-library
[2] https://docs.plone.org/external/plone.app.robotframework/docs/source/robot.html

**Comment by michalanglart** [comment]
Hi,

Do you edit big files? We do notice UI slowness but for files having e.g. 8000 lines (see #259). Do you notice RED not being greatly responsive overall or only in editor? What's your machine hardware?

As to the latter: we were discussing Library imports, not keyword usage. The "keyword from nested library" is being reported when:

- we have a resource file b.robot which in it's setting section imports e.g. library X
- we have another file (a.robot, suite or resource) which imports b.robot in Resource setting
- inside a.robot we are using keyword Y from library X

In setup as above a.robot is importing library X but not "explicitly" via Library setting, but transitively because the resource file it imports is importing this library ("implicitly").

**Comment by humbienri** [comment]
Hello again,

No I am not editing big files.  Not big at all as they are pretty new and I'm the only one that's been working on them.  No more than 200 LOCs I'd say.  The system I am referring to is a Windows 10 laptop, 64bit OS, 16GBs of RAM if I remember correctly.

It's not a super beast in terms of hardware but it's also not an old laptop like my personal one.  One of the reasons I keep going back to VSCode for editing some of my *.robot files is because VSCode is so much more responsive compared to RED, but I do like RED.

Thanks again @michalanglart!

**Comment by balaji-githubstore** [comment]
![Screenshot (340)](https://user-images.githubusercontent.com/31404824/87167868-45a89280-c2eb-11ea-9bf2-e06e892e242f.png)
